src/NHibernate/Cfg/SettingsFactory.cs

```diff
@@ -395,38 +385,45 @@ private static System.Type CreateLinqQueryProviderType(IDictionary<string, strin
 
 		private static ITransactionFactory CreateTransactionFactory(IDictionary<string, string> properties)
 		{
-			string className = PropertiesHelper.GetString(
-				Environment.TransactionStrategy, properties, typeof(AdoNetWithSystemTransactionFactory).FullName);
-			log.Info("Transaction factory: {0}", className);
-
-			try
-			{
-				var transactionFactory =
-					(ITransactionFactory)
-					Environment.ObjectsFactory.CreateInstance(ReflectHelper.ClassForName(className));
-				transactionFactory.Configure(properties);
```

Comment:
This line is missing, it must be put back after the call to `PropertiesHelper.GetInstance`.